### PR TITLE
FUSETOOLS-2686 - provide fis for fuse7 template

### PR DESCRIPTION
**Pull request by apupier**
it will require to update the name of FIS image (in case https://issues.jboss.org/browse/OSFUSE-683 is really done)
--> name updated (even if still cannot test as registry https://registry-console.engineering.redhat.com/ is nto reachable)

There are 2 commits:
- one to add a FIS Template for Fuse 7 with static fabric8-maven-plugin
- one to make resolution dynamic of the fabric8-maven-plugin, it is required due to the fact that we will know the versions to use only after the release is done. previously, it was based on one fabric8-maven-plugin and it was working with newer version of the bom.

# Pull Request Checklist

After this checklist is all checked or PR provides explanations for possible pass-through, please put the label "Ready for review" on the PR.

## General

- [x] Did you use the Jira Issue number in the commit comments?
- [x] Did you set meaningful commit comments on each commit?
- [x] Did you sign off all commits for this PR? (git commit -s -m "jira issue number - your commit comment")

## Functional

- [x] Did the CI job report a successful build?
- [x] Is the non-happy path working, too?

## Maintainability

- [x] Are all Sonar reported issues fixed or can they be ignored?
- [x] Is there no duplicated code?
- [x] Are method-/class-/variable-names meaningful?

## Tests

- [x] Are there unit-tests? --> only an integration test just checking that the project is created (as for previous versions of the template)
- [x] Are there integration tests (or at least a jira to tackle these)?

## Legal

- [x] Have you used the correct file header copyright comment?
- [x] Have you used the correct year in the headers of new files?

### CODE REVIEW

**Comment by apupier** [comment]
to test it, need to change the name of the image as it is available only on staging repo:
-Dfabric8.generator.from=docker-registry.engineering.redhat.com/jboss-fuse-7/fis-java-openshift:1.0

(or -Dfabric8.generator.from=docker-registry.engineering.redhat.com/jboss-fuse-7/fuse-java-openshift:1.0 if already pushed)

**Comment by apupier** [comment]
edit: can be tested with the new URL as the image has been pushed with the new name:
https://registry-console.engineering.redhat.com/registry#/images/jboss-fuse-7/fuse-java-openshift:1.0

File: editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/impl/simple/OSESpringBootXMLTemplateForFuse7.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Red Hat, Inc.
```
Comment (lhein) [inline editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/impl/simple/OSESpringBootXMLTemplateForFuse7.java]: 2018

File: editor/plugins/org.fusesource.ide.projecttemplates/template-sources/simple/ose/simple-ose-log-springboot-fuse7/configuration/settings.xml [hunk 2]
```diff
@@ -0,0 +1,235 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+     Copyright 2005-2016 Red Hat, Inc.
```
Comment (lhein) [inline editor/plugins/org.fusesource.ide.projecttemplates/template-sources/simple/ose/simple-ose-log-springboot-fuse7/configuration/settings.xml]: 2018

File: editor/plugins/org.fusesource.ide.projecttemplates/template-sources/simple/ose/simple-ose-log-springboot-fuse7/src/main/java/org/mycompany/Application.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
+/**
+ *  Copyright 2005-2017 Red Hat, Inc.
```
Comment (lhein) [inline editor/plugins/org.fusesource.ide.projecttemplates/template-sources/simple/ose/simple-ose-log-springboot-fuse7/src/main/java/org/mycompany/Application.java]: 2018

File: targetplatform/without spaceJBossTools Oxygen Windows x86_64 (1).launch [hunk 4]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (lhein) [inline targetplatform/without spaceJBossTools Oxygen Windows x86_64 (1).launch]: is that file really needed?

File: editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/impl/simple/OSESpringBootXMLTemplateForFuse7.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+	}
+
+	@Override
+	public TemplateCreatorSupport getCreator(CommonNewProjectMetaData projectMetaData) {
+		return new OSEUnzipTemplateCreator("7");
+	}
+
+	@Override
+	public boolean isCompatible(String camelVersion) {
+		return new ComparableVersion(camelVersion).compareTo(minimalCompatibleCamelVersion) >= 0;
```
Comment (lhein) [inline editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/impl/simple/OSESpringBootXMLTemplateForFuse7.java]: how about reusing https://github.com/jbosstools/jbosstools-fuse/blob/master/editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/util/BasicProjectCreatorRunnableUtils.java#L92 to not duplicate code

File: editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/impl/simple/OSESpringBootXMLTemplateForFuse6.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+public class OSESpringBootXMLTemplateForFuse6 extends AbstractOSESpringBootXMLTemplate {
+	
+	private static ComparableVersion minimalCompatibleCamelVersion = new ComparableVersion("2.18.0");
+	private static ComparableVersion maximalCompatibleCamelVersion = new ComparableVersion("2.20.0");
+	
+	@Override
+	public boolean isCompatible(String camelVersion) {
+		return new ComparableVersion(camelVersion).compareTo(minimalCompatibleCamelVersion) >= 0
+				&& new ComparableVersion(camelVersion).compareTo(maximalCompatibleCamelVersion) < 0;
```
Comment (lhein) [inline editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/impl/simple/OSESpringBootXMLTemplateForFuse6.java]: How about reusing https://github.com/jbosstools/jbosstools-fuse/blob/master/editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/util/BasicProjectCreatorRunnableUtils.java#L92 to not duplicate code too much

File: editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/adopters/AbstractProjectTemplate.java [hunk 7]
```diff
@@ -99,12 +99,12 @@ protected boolean isStrictlyLowerThan2200(String camelVersion) {
 		return new ComparableVersion(camelVersion).compareTo(COMPARABLE_CAMEL_2_20_0_VERSION) < 0;
```
Comment (lhein) [inline editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/adopters/AbstractProjectTemplate.java]: https://github.com/jbosstools/jbosstools-fuse/blob/master/editor/plugins/org.fusesource.ide.projecttemplates/src/org/fusesource/ide/projecttemplates/util/BasicProjectCreatorRunnableUtils.java#L92

File: targetplatform/without spaceJBossTools Oxygen Windows x86_64 (1).launch (hunk 4 above)
Comment (apupier) [reply targetplatform/without spaceJBossTools Oxygen Windows x86_64 (1).launch]: not at all

**Comment by apupier** [comment]
testPR
